docs/api-communication-spec.md

```diff
@@ -0,0 +1,686 @@
+# PCAP Release API Documentation
+
+This document contains the API specification for the PCAP release.
+
+The pcap-release automates the identification of network capture targets based on metadata from:
+- **Cloud Foundry Applications**, where network traffic is captured directly from the application container.  
+- **BOSH** Deployment VMs, where network traffic is captured on the VM.
+
+Pcap-release uses gRPC for communication between all components. This document highlights the intended use of fields, scenarios considered by the implementation and detailed explanation of the data structures.
+
+## CF Case Overview
+
+Capturing data from cloud foundry applications requires the pcap-agent inside the app container. The pcap-agent runs in plain-text HTTP/2 (H2C). The Envoy in the app container provides the TLS termination.
+
+The Cloud Controller is used to identify the target ports of the pcap-agent in the target app instance.
```

Comment:
It depends, if we go full app-ssh-style we would use routes and nats to find the relevant ports, not the CC